### DISCUSSION TITLE
Issues embedding Excalidraw into ReactFlow.

### DISCUSSION THREAD

**Discussion by gdaddi** [excalidraw/excalidraw#8744] [Q&A]
Hey! 

Has anyone managed to embed an Excalidraw component into a ReactFlow custom node? I have been trying for a while but keep getting very weird visual bugs that I think are caused by coordinates transformations introduced by Reactflow since I have no issue embedding Excalidraw components in any other div.

I tried all sorts of styling changes/coordinate transformations, but keep getting these issues:
* When I create an element in Excalidraw, the position of my mouse is different from the position where the Excalidraw element is created.  
* I often see the Excalidraw canvas outside of the cusom node bounds, almost as if it was ignoring the `inset-0` tailwind directive. 

Any insight about what could be causing this would be very useful!

This is a sample page to replicate the issue.
```js
'use client';

import React, { useCallback, useState } from 'react';
import dynamic from 'next/dynamic';
import {
  ReactFlow,
  applyNodeChanges,
  Background,
} from '@xyflow/react';
import '@xyflow/react/dist/style.css';
import { BackgroundVariant } from '@xyflow/react';


// Dynamic import of Excalidraw to handle SSR
const Excalidraw = dynamic(
  async () => (await import("@excalidraw/excalidraw")).Excalidraw,
  { ssr: false }
);

// Custom Node component with Excalidraw embedded
const CustomDrawingNode = ({ data, isConnectable }) => {
  return (
    <div
      className="text-updater-node relative rounded-md bg-gray-100 dark:bg-gray-800"
      style={{
        width: '1000px',
        height: '500px',
      }}
    >
      <div className="absolute inset-0">
        <Excalidraw
          initialData={{
            elements: [],
            appState: {
              viewBackgroundColor : '#fafafa',
              currentItemStrokeColor: '#000000',
              theme: 'light',
            },
          }}
          gridModeEnabled={false}
          zenModeEnabled={true}
          viewModeEnabled={false}
        />
      </div>
    </div>
  );
};

// Main Flow component for ReactFlow setup
const FlowPage = () => {
  const [nodes, setNodes] = useState([
    {
      id: 'node-1',
      type: 'customDrawing',
      position: { x: 1000, y: 5 },
      data: { prefersDarkMode: false },
      draggable: false,
    },
  ]);
  const [edges, setEdges] = useState([]);

  const onNodesChange = useCallback(
    (changes) => setNodes((nds) => applyNodeChanges(changes, nds)),
    [setNodes],
  );

  return (
    <div style={{ width: '100vw', height: '100vh' }}>
      <ReactFlow
        nodes={nodes}
        edges={edges}
        onNodesChange={onNodesChange}
        nodeTypes={{ customDrawing: CustomDrawingNode }}
        panOnDrag={false}
        zoomOnScroll={false}
        fitView
        style={{ backgroundColor: '#B8CEFF' }}
      >
        <Background color="#B8CEFF" gap={16} size={2} variant={BackgroundVariant.Lines} />
      </ReactFlow>
    </div>
  );
};

export default FlowPage;

```

**Comment by aayush7511** [comment]
Hi, 
You are using ReactFlow which could be causing your inset-0 to behave differently. I think you had the right idea to use inline styles here:
 `<div
      className="text-updater-node relative rounded-md bg-gray-100 dark:bg-gray-800"
      style={{
        width: '1000px',
        height: '500px',
      }}
    >`
    
since Excalidraw uses its parent's dimension to render the canvas. However, try making the height and width of the immediate parent 1000px and 500px respectively. 
`<div style={{
        width: '1000px',
        height: '500px',
      }}>
        <Excalidraw
          initialData={{
            elements: [],
            appState: {
              viewBackgroundColor : '#fafafa',
              currentItemStrokeColor: '#000000',
              theme: 'light',
            },
          }}
          gridModeEnabled={false}
          zenModeEnabled={true}
          viewModeEnabled={false}
        />
      </div>`
